pkg/virt-controller/watch/drain/evacuation/evacuation.go

```diff
@@ -376,19 +376,29 @@ func (c *EvacuationController) sync(node *k8sv1.Node, vmisOnNode []*virtv1.Virtu
 
 	migrationCandidates, nonMigrateable := c.filterRunningNonMigratingVMIs(vmisToMigrate, activeMigrations)
 
+	activeMigrationsFromThisSourceNode := c.numOfVMIMForThisSourceNode(vmisOnNode, activeMigrations)
+
+	maxParallelMigrationsPerOutboundNode :=
+		int(*c.clusterConfig.GetMigrationConfiguration().ParallelOutboundMigrationsPerNode)
 	// Don't create hundreds of pending migration objects.
 	// This is just best-effort and is *not* intended to not overload the cluster.
 	// It is possible that more migrations than the limit are created because of evacuations on other nodes.
 	// The migration controller needs to limit itself to a reasonable number of running migrations
 	maxParallelMigrations := int(*c.clusterConfig.GetMigrationConfiguration().ParallelMigrationsPerCluster)
-	if len(activeMigrations) >= maxParallelMigrations {
-		// We have to re-enqueue if some work is left, since migrations from other controllers or workers` don't wake us up again
-		if len(migrationCandidates) > 0 || len(nonMigrateable) > 0 {
+
+	// No free slots, need to wait till migrations will finish. Re-enqueue and return
+	if len(migrationCandidates) > 0 || len(nonMigrateable) > 0 {
```

Comment:
Don't we handle the `if len(migrationCandidates) > 0 || len(nonMigrateable) > 0 ` on line 372?

```
vmisToMigrate := vmisToMigrate(node, vmisOnNode, taint)
	if len(vmisToMigrate) == 0 {
		return nil
	}
```

Reply:
@xpivarc `vmisToMigrate` is a list of VMs marked for eviction because of their eviction strategy. `migrationCandidates` are VMs that marked and meet the conditions for live migration. `nonMigrateable` are VMs  that were marked but don't meet the requirements for live migration